crates/polars-io/src/parquet/write.rs

```diff
@@ -191,8 +191,8 @@ where
         // ensures all chunks are aligned.
         df.align_chunks();
 
-        let n_splits = df.height() / self.row_group_size.unwrap_or(512 * 512);
-        if n_splits > 0 {
+        let n_splits = self.row_group_size.map(|s| df.height() / s).unwrap_or(1);
+        if n_splits > 1 {
```

Comment:
Not sure if this change is desired, do we need potentially need this to align chunks? If I only have a single chunk, does `accumulate_dataframes_vertical_unchecked` create a copy of the data?

Reply:
Now that I see this. I think we should make this work on an immutable `&df`. As we now write back the chunks to the owner. Whereas it isn't preferable to go from non-chunked data to chunked data.